scripts/gulp/tasks/gulpWebpack.ts

```diff
@@ -16,10 +19,27 @@ type RunWebpackCfg = {
   cwd: string
   prefix: string
   args?: string[]
-  env?: object
+  env?: Env
   devServer?: boolean
 }
 
+// https://github.com/cypress-io/cypress/issues/18914
+// Node 17+ ships with OpenSSL 3 by default, so we may need the option
+// --openssl-legacy-provider so that webpack@4 can use the legacy MD4 hash
+// function. This option doesn't exist on Node <17 or when it is built
+// against OpenSSL 1, so we have to detect Node's major version and check
+// which version of OpenSSL it was built against before spawning the process.
+//
+// Can be removed once the webpack version is upgraded to >= 5.61,
+// which no longer relies on Node's builtin crypto.hash function.
+function useLegacyOpenSSLProvider (env: Env) {
+  if (process.versions && semver.satisfies(process.versions.node, '>=17.0.0') && process.versions.openssl.startsWith('3.')) {
```

Comment:
Could use `semver` here, something like `semver.satisifies(prices.versions, '>=3')`.

Reply:
Works with a tweak since Node 17.0.0 uses openssl=`3.0.0-quic` which is considered a pre-release so I've included `{ includePrerelease: true }`.